### PR TITLE
Refactor pom structure

### PR DESCRIPTION
**Pull request by jianghaolu**
- The original parent for all management libraries `pom.xml` is now `pom.management.xml`.
- The original parent for all track 1 and track 2 libraries `pom.client.xml` is now `pom.client.xml` and `pom.data.xml`.
- The new `pom.xml` is an aggregator containing all the projects. 
- The new `parent/pom.xml` contains all the version and dependency definitions, plus individual service level plugin definitions.

The build command for everything is `mvn -Dgpg.skip -DskipTests install`. The build command for Track 2 libraries to generate spotbugs reports is `mvn -Dgpg.skip -DskipTests -Dinclude-non-shipping-modules install -f pom.client.xml`. The build command to build all management libraries currently in a Azure Stack profile is `mvn -Dgpg.skip -DskipTests install -f pom.management.xml`. The build command to build all data plane libraries with spotbugs reports is `mvn -Dgpg.skip -DskipTests -Dinclude-non-shipping-modules -Ddata-plane install -f pom.data.xml`.

### CODE REVIEW

**Comment by jianghaolu** [comment]
Travis is complaining gpg key not found. I can fix it with `-Dgpg.skip` but should we make that the default behavior? I have not used gpg once and the developer experience (git clone, maven install -DskipTests) is bad (fail on Gpg).

**Comment by weshaggard** [comment]
cc @Azure/azure-sdk-eng 
PTAL @mitchdenny

File: .azure-pipelines/client.yml [hunk 1]
```diff
@@ -97,6 +97,18 @@ jobs:
...
         displayName: 'Download dependency checker'
 
+      - task: Maven@3
+        displayName: 'Install azure-sdk-parent'
+        inputs:
+          mavenPomFile: parent/pom.xml
+          options: '$(DefaultOptions) -DskipTests -Dgpg.skip'
+          mavenOptions: '$(LoggingOptions)'
+          javaHomeOption: 'JDKVersion'
+          jdkVersionOption: '1.11'
```
Comment (sima-zhu) [inline .azure-pipelines/client.yml]: We moved to JAVA 11 for track 2?

Comment (jianghaolu) [reply .azure-pipelines/client.yml]: I think it's just the azure pipeline build setting.

File: pom.xml [hunk 2]
```diff
@@ -22,389 +31,247 @@
...
-                    <resource>
-                      <directory>../extlib</directory>
-                    </resource>
-                 </resources>
-              </configuration>
-           </execution>
+      <!-- Copy the pom file to output -->
+      <plugin>
+        <artifactId>maven-antrun-plugin</artifactId>
+        <version>1.8</version>
```
Comment (sima-zhu) [inline pom.xml]: Better to have variable.

File: pom.management.xml [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+          <execution>
+            <phase>none</phase>
+          </execution>
+        </executions>
+      </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.1</version>
```
Comment (sima-zhu) [inline pom.management.xml]: Using variable for version.

File: pom.management.xml [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-release-plugin</artifactId>
+        <version>2.5.3</version>
+      </plugin>
+
+     <plugin>
+       <groupId>com.googlecode.addjars-maven-plugin</groupId>
+       <artifactId>addjars-maven-plugin</artifactId>
+       <version>1.0.5</version>
```
Comment (sima-zhu) [inline pom.management.xml]: Same here

File: pom.management.xml [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+      <id>run-shade-plugin</id>
+      <activation>
+        <property><name>runShadePlugin</name></property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-shade-plugin</artifactId>
+            <version>2.4.1</version>
```
Comment (sima-zhu) [inline pom.management.xml]: Same

File: pom.data.xml [hunk 6]
```diff
@@ -0,0 +1,550 @@
...
+            <reports>
+              <report>aggregate</report>
+            </reports>
+          </reportSet>
+        </reportSets>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <version>3.0.0</version>
```
Comment (sima-zhu) [inline pom.data.xml]: Use variable.

File: pom.xml (hunk 2 above)
Comment (jianghaolu) [reply pom.xml]: I just copied from the current pom.client.xml but I'll fix it.

File: pom.client.xml [hunk 7]
```diff
@@ -1064,7 +517,7 @@
     </profile>
 
     <profile>
-      <id>non-shipping-modules</id>
+      <id>non-shippping-modules</id>
```
Comment (JonathanGiles) [inline pom.client.xml]: Renaming here is adding a typo

Comment (jianghaolu) [reply pom.client.xml]: Fixed

File: pom.data.xml [hunk 8]
```diff
@@ -0,0 +1,550 @@
...
+        </property>
+      </activation>
+      <modules>
+        <module>./sdk/template/azure-sdk-template</module>
+      </modules>
+    </profile>
+  </profiles>
+
+  <modules>
+    <module>./batch/data-plane</module>
```
Comment (weshaggard) [inline pom.data.xml]: Shouldn't these be updated to the new parent azure-data-sdk-parent?

Comment (jianghaolu) [reply pom.data.xml]: Forgot that. Thanks for catching it!

**Pull request merged**